### PR TITLE
[New Rule] Network Activity Detected via kworker

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
This rule monitors for network connections from a kworker process. kworker, or kernel worker, processes are part of the kernel's workqueue mechanism. They are responsible for executing work that has been scheduled to be done in kernel space, which might include tasks like handling interrupts, background activities, and other kernel-related tasks. Attackers may attempt to evade detection by masquerading as a kernel worker process.

## RTA and Data
The RTA and Data is currently available in a direct PR to react. This is due to a CIDR match bug. The RTA & data can be found in: https://github.com/elastic/react/pull/113

### Detection
All examples in the screenshot below are examples of malware setting up network connections. 1 system in telemetry shows this behavior while connecting to a file share. 

Converted to DR with new_terms instead of ER DIAG to check for TP/FP ratio. Will tune and potentially convert to ER in the future.

```
host.os.type:linux and event.category:network and event.action:(connection_attempted or connection_accepted) and 
process.name:kworker*
```
image

### CODE REVIEW

**Review by eric-forte-elastic — APPROVED** [review]
:green_circle: Manual review, looks good to me! :+1:

File: rules/linux/command_and_control_linux_kworker_netcon.toml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        "Use Case: Threat Detection",
+        "Tactic: Command and Control",
+        "Data Source: Elastic Defend"
+        ]
+timestamp_override = "event.ingested"
+type = "new_terms"
+
+query = '''
+host.os.type:linux and event.category:network and event.action:(connection_attempted or connection_accepted) and 
+process.name:kworker*
```
Comment (Aegrah) [inline rules/linux/command_and_control_linux_kworker_netcon.toml:37]:
```suggestion
process.name:kworker* and destination.ip != null and 
not cidrmatch(destination.ip, "127.0.0.0/8", "169.254.0.0/16", "224.0.0.0/4", "::1")
```

File: rules/linux/command_and_control_linux_kworker_netcon.toml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        "Use Case: Threat Detection",
+        "Tactic: Command and Control",
+        "Data Source: Elastic Defend"
+        ]
+timestamp_override = "event.ingested"
+type = "new_terms"
+
+query = '''
+host.os.type:linux and event.category:network and event.action:(connection_attempted or connection_accepted) and 
+process.name:kworker* and destination.ip != null and
```
Comment (Aegrah) [inline rules/linux/command_and_control_linux_kworker_netcon.toml:37]:
```suggestion
process.name:kworker* and destination.ip : * and 
```

File: rules/linux/command_and_control_linux_kworker_netcon.toml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        "Use Case: Threat Detection",
+        "Tactic: Command and Control",
+        "Data Source: Elastic Defend"
+        ]
+timestamp_override = "event.ingested"
+type = "new_terms"
+
+query = '''
+host.os.type:linux and event.category:network and event.action:(connection_attempted or connection_accepted) and 
+process.name:kworker* and destination.ip : * and
```
Comment (Aegrah) [inline rules/linux/command_and_control_linux_kworker_netcon.toml:37]:
```suggestion
process.name:kworker* and destination.ip : "*" and 
```

File: rules/linux/command_and_control_linux_kworker_netcon.toml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+type = "new_terms"
+
+query = '''
+host.os.type:linux and event.category:network and event.action:(connection_attempted or connection_accepted) and 
+process.name:kworker*
+'''
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+
```
Comment (w0rk3r) [inline rules/linux/command_and_control_linux_kworker_netcon.toml:42]: Would be nice if we could find techniques to include here and in the Exfiltration tactic

**Review by w0rk3r — APPROVED** [review]
🧙🏼

**Pull request merged**